zerver/lib/user_groups.py

```diff
@@ -125,6 +126,75 @@ def access_user_group_by_id(
     return user_group
 
 
+def access_user_group_for_deactivation(
+    user_group_id: int, user_profile: UserProfile
+) -> NamedUserGroup:
+    user_group = access_user_group_by_id(user_group_id, user_profile, for_read=False)
+
+    if (
+        user_group.direct_supergroups.exclude(named_user_group=None)
+        .filter(named_user_group__deactivated=False)
+        .exists()
+    ):
+        raise JsonableError(
+            _("You cannot deactivate a user group that is subgroup of any user group.")
+        )
+
+    anonymous_supergroup_ids = user_group.direct_supergroups.filter(
+        named_user_group=None
+    ).values_list("id", flat=True)
+
+    setting_group_ids_using_deactivating_user_group = [
+        *list(anonymous_supergroup_ids),
+        user_group.id,
+    ]
```

Comment:
This next section probably deserves a comment about how checking whether the group is unused requires checking both whether the group is used directly as the value of a setting or as a subgroup of an anonymous group used for such a setting.

Reply:
Added the comment.